### ISSUE TITLE
oci_pull tag non-reproducibility warning is misleading

### ISSUE THREAD

**Issue by nacl** [bazel-contrib/rules_oci#188]
When experimenting with rules_oci, I eventually came across warnings like this when just using `tag` in `oci_pull`:

```
WARNING: for reproducible builds, a digest is recommended.
Either set 'reproducible = False' to silence this warning,
or run the following command to change oci_pull to use a digest:

bazel run @ubuntu_unpinned//:pin
```

The code for this seems to be coming from here:

https://github.com/bazel-contrib/rules_oci/blob/82d5b6ab077f689485d9631c4744b424be30b7d2/oci/pull.bzl#L109-L121

This warning makes perfect sense, but it being a warning suggests that the `@ubuntu` repo will still be initialized even if I don't pin it or add `reproducible = False` to my `oci_pull` repository rule.  From the line at the bottom of this, it isn't, and leads to future errors like:

```
ERROR: /work/BUILD.bazel:9:10: no such package '@ubuntu//': The repository '@ubuntu' could not be resolved: Repository '@ubuntu' is not defined and referenced by '//:image'
```

Please consider updating this.  Perhaps this message should be printed with `fail` instead?

**Comment by thesayyn** [comment]
It makes sense. I tried to tackle this in #105. I'll give it another stab.
